docs/source/running-jobs/environment-variables.rst

```diff
@@ -0,0 +1,110 @@
+
+.. _env-vars:
+
+Using Environment Variables
+================================================
+
+User-specified environment variables
+------------------------------------------------------------------
+
+You can specify environment variables to be made available to a task in two ways:
+
+- The ``envs`` field (dict) in a :ref:`task YAML <yaml-spec>`
+- The ``--env`` flag in the ``sky launch/exec`` :ref:`CLI <cli>` (takes precedence over the above)
+
+The ``file_mounts``, ``setup``, and ``run`` sections of a task YAML file can then access these variables via the ``${MYVAR}`` syntax.
+
+Using in ``file_mounts``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+    envs:
+      MY_BUCKET: skypilot-temp-gcs-test
+      MY_LOCAL_PATH: tmp-workdir
+      MODEL_SIZE: 13b
+
+    file_mounts:
+        /mydir:
+            name: ${MY_BUCKET}  # Name of the bucket.
+            mode: MOUNT
+
+        /another-dir2:
+            name: ${MY_BUCKET}-2
+            source: ["~/${MY_LOCAL_PATH}"]
+
+        /checkpoint/${MODEL_SIZE}: ~/${MY_LOCAL_PATH}
+
+The values of these variables are filled in by SkyPilot at task YAML parse time.
+
+Read more at `examples/using_file_mounts_with_env_vars.yaml <https://github.com/skypilot-org/skypilot/blob/master/examples/using_file_mounts_with_env_vars.yaml>`_.
+
+Using in ``setup``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+All user-specified environment variables are exported to a task's ``setup`` commands.
+
+Using in ``run``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+All user-specified environment variables are exported to a task's execution (i.e., while its ``run`` commands are running).
```

Comment:
Maybe we can combine the two sections into one to reduce the duplication.
```
Using in ``setup`` and ``run``
```

Reply:
Good call, done.